### ISSUE TITLE
Pose2Pose

### ISSUE THREAD

**Issue by rainofmine** [ShenhanQian/SpeechDrivesTemplates#12]
I tried the pose2pose training, but the loss seems never converge. And the pose reconstruction is not correct. Dose anything wrong?

**Comment by ShenhanQian** [comment]
What data do you use for training?

**Comment by rainofmine** [comment]
> What data do you use for training?

Oliver. I just run it as you proposed in README. And the SDT-BP training is correct. So I think the data has no problem.

**Comment by ShenhanQian** [comment]
Thanks for reporting this! 

After testing from our side, we conclude that this is caused by a false hyperparameter. `POSE2POSE.LAMBDA_KL` was accidentally set to `1.0`, which prevents the regression loss from converging.

The right value should be `0.1`, which is just the value we used for experiments in our paper. We have corrected the code ([here](https://github.com/ShenhanQian/SpeechDrivesTemplates/blob/3036ee908ae2b6098327916d843c7acfa45a0755/configs/default.py#L43)). Please pull the newest commit.

**Comment by rainofmine** [comment]
> Thanks for reporting this!
> 
> After testing from our side, we conclude that this is caused by a false hyperparameter. `POSE2POSE.LAMBDA_KL` was accidentally set to `1.0`, which prevents the regression loss from converging.
> 
> The right value should be `0.1`, which is just the value we used for experiments in our paper. We have corrected the code ([here](https://github.com/ShenhanQian/SpeechDrivesTemplates/blob/3036ee908ae2b6098327916d843c7acfa45a0755/configs/default.py#L43)). Please pull the newest commit.

I tried the new hyperparameter. The regression loss only declined from ~0.7 to ~0.5. And the reconstruction result is not good. So what is the expected result？

**Comment by ShenhanQian** [comment]
You should not expect a precise reconstruction with pose2pose training because the audio is not used. If you check the saved videos, you will observe that the reconstructed gesture sequence on the left is very like the real one on the right in general while lacking details, which is just the thing we call "template" in our paper.

To help validate your experiment, the below is what I get after training 100 epochs on Oliver:
```
2022-04-27 10:20:13,042 [INFO] [VAL] epoch: 100/100  val_time: 0.7 min  num_samples: 10546  reg_loss: 0.43062  kl_loss: 0.08418  loss: 0.51480  L2_dist: 26.18086  lip_sync_error_n: 0.20510
```
Therefore, what you have observed is normal.

You may refer to the section from 4:44 - 4:54 in [this](https://youtu.be/yu-5gUHn6h8) video to know what a template gesture sequence should look like.

**Comment by rainofmine** [comment]
> You should not expect a precise reconstruction with pose2pose training because the audio is not used. If you check the saved videos, you will observe that the reconstructed gesture sequence on the left is very like the real one on the right in general while lacking details, which is just the thing we call "template" in our paper.
> 
> To help validate your experiment, the below is what I get after training 100 epochs on Oliver:
> 
> ```
> 2022-04-27 10:20:13,042 [INFO] [VAL] epoch: 100/100  val_time: 0.7 min  num_samples: 10546  reg_loss: 0.43062  kl_loss: 0.08418  loss: 0.51480  L2_dist: 26.18086  lip_sync_error_n: 0.20510
> ```
> 
> Therefore, what you have observed is normal.
> 
> You may refer to the section from 4:44 - 4:54 in [this](https://youtu.be/yu-5gUHn6h8) video to know what a template gesture sequence should look like.

Thank you for the explanation.

**Comment by rainofmine** [comment]
Thank you for the explanation.
